spec/controllers/tasks_controller_spec.rb

```diff
@@ -74,6 +74,15 @@
         end
       end
     end
+
+    context 'when no user is signed in' do
+      before { sign_out user }
+
+      it 'redirects to root page' do
+        get_request
+        expect(response).to redirect_to(root_path)
```

Comment:
I would also consider checking the flash message.

As a user, it is important to know, the reason behind a "failure". For that reason, the existence of the message is essential.

Reply:
Sure, this makes sense here as well. I just added a corresponding test case.